navigation/pose_action_server/src/pose_action_server_ros.cpp

```diff
@@ -0,0 +1,148 @@
+#include <pose_action_server/pose_action_server_ros.hpp>
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+
+PoseActionServerNode::PoseActionServerNode() : Node("pose_action_server_node") {
+    auto pose_sub_topic =
+        this->declare_parameter<std::string>("pose_sub_topic");
+    auto action_name = this->declare_parameter<std::string>("action_name");
+
+    rclcpp::QoS qos = rclcpp::QoS(rclcpp::KeepLast(10))
+                          .reliability(RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT);
+
+    pose_sub_ = this->create_subscription<geometry_msgs::msg::PoseStamped>(
+        pose_sub_topic, qos,
+        std::bind(&PoseActionServerNode::pose_callback, this, _1));
+
+    action_server_ =
+        rclcpp_action::create_server<vortex_msgs::action::FilteredPose>(
+            this, action_name,
+            std::bind(&PoseActionServerNode::handleGoal, this, _1, _2),
+            std::bind(&PoseActionServerNode::handleCancel, this, _1),
+            std::bind(&PoseActionServerNode::handleAccepted, this, _1));
+}
+
+rclcpp_action::GoalResponse PoseActionServerNode::handleGoal(
+    const rclcpp_action::GoalUUID& uuid,
+    std::shared_ptr<const vortex_msgs::action::FilteredPose::Goal> goal) {
+    RCLCPP_INFO(this->get_logger(), "Received request to filter pose.");
+    (void)uuid;
```

Comment:
If you have the `(void)uuid` here because the compiler complains about unused variables, you can instead use unnamed input parameters

i.e. 

```C++
rclcpp_action::GoalResponse PoseActionServerNode::handleGoal(
    const rclcpp_action::GoalUUID& /*uuid*/,
    std::shared_ptr<const vortex_msgs::action::FilteredPose::Goal> goal) 
```

The `/*uuid*/` part isnt strictly needed but its nice to leave it in there for future reference